pharus/component_interface.py

```diff
@@ -114,8 +109,167 @@ def restriction(self):
+                        field_name_map[
+                            field["input"] if "input" in field else field["destination"]
+                        ] = field["destination"]
+                    else:
+                        for submap in field["map"]:
+                            field_name_map[
+                                submap["input"]
+                                if "input" in submap
+                                else submap["destination"]
+                            ] = submap["destination"]
+                self.payload = {field_name_map[k]: v for k, v in self.payload.items()}
+            for t in self.tables:
+                t.insert1(
+                    {k: v for k, v in self.payload.items() if k in t.heading.attributes}
+                )
+        return "Insert successful"
+
+    def fields_route(self):
+        fields = []
+        if self.fields_map:
+            datatype_map = self._get_datatype_map()
+            for field in self.fields_map:
+                field_type = field["type"]
+                field_name = (
+                    field["input"] if "input" in field else field["destination"]
+                )
+                if field_type == "attribute":
+                    field_datatype = datatype_map[field["destination"]]
+                    fields.append(
+                        dict(type=field_type, datatype=field_datatype, name=field_name)
+                    )
+                elif field_type == "table":
+                    s, t = field["destination"].split(".")
+                    if "map" in field:
+                        proj_map = {
+                            attr["input"]
+                            if "input" in attr
+                            else attr["destination"]: attr["destination"]
+                            for attr in field["map"]
+                        }
+                        records = (
+                            getattr(
+                                dj.VirtualModule(
+                                    s,
+                                    s,
+                                    connection=self.connection,
+                                ),
+                                t,
+                            )
+                            .proj(**proj_map)
+                            .fetch("KEY")
+                        )
+                    else:
+                        records = getattr(
+                            dj.VirtualModule(
+                                s,
+                                s,
+                                connection=self.connection,
+                            ),
+                            t,
+                        ).fetch("KEY")
+                    fields.append(
+                        dict(type=field_type, values=records, name=field_name)
+                    )
+        else:
+            for t in self.tables:
+                for v in t.heading.attributes.values():
+                    is_fk = False
```

Comment:
Flags like this are unnecessary if you use a variable's *truthy* characteristic.